File: ejercicio4/euclides.c
#include <stdio.h>
#include <time.h>

int mcd(int a, int b) {
    if (b == 0)
        return a;
    return mcd(b, a % b);
}

int main() {
    int a = 123456789, b = 987654321;
    int resultado;

    clock_t inicio = clock();

    for (int i = 0; i < 10000000; i++) {
        resultado = mcd(a, b);
    }

    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;

    printf("MCD: %d\n", resultado);
    printf("Tiempo: %f segundos\n", tiempo);

    return 0;
}

File: ejercicio4/euclides.hs
import System.CPUTime
import Text.Printf

mcd :: Integer -> Integer -> Integer
mcd a 0 = a
mcd a b = mcd b (a `mod` b)

main = do
    start <- getCPUTime
    let resultado = last [mcd 123456789 987654321 | _ <- [1..10000000]]
    end <- getCPUTime

    let tiempo = fromIntegral (end - start) / (10^12)

    printf "MCD: %d\n" resultado
    printf "Tiempo: %f segundos\n" (tiempo :: Double)
